plugins/autolink/plugin.js

```diff
@@ -38,7 +38,11 @@
 				return;
 			}
 
-			var commitKeystrokes = editor.config.autolink_commitKeystrokes || CKEDITOR.config.autolink_commitKeystrokes;
+			// (#3156)
+			var commitKeystrokes = editor.config.autolink_commitKeystrokes || CKEDITOR.config.autolink_commitKeystrokes,
+				emailRegex = editor.config.autolink_emailRegex || CKEDITOR.config.autolink_emailRegex,
+				urlRegex = editor.config.autolink_urlRegex || CKEDITOR.config.autolink_urlRegex;
```

Comment:
Indeed, using editor config fixes the issue. Actually, `editor.config` will be populated automatically by editor itself with `CKEDITOR.config` if no option has been directly set on `editor.config.`. That means we could safely remove "or" (`||`) condition for regex aliases (and commit keystrokes also).